### DISCUSSION TITLE
std::span as an datatype for messages

### DISCUSSION THREAD

**Discussion by SpiRelli90** [uNetworking/uWebSockets#1413] [Ideas]
Hello, 
first of all thank you for your work on this library! 
Currently every message to send or to publish uses string_view. I guess this is because it is a non owning continious block of memory.
Wouldn't the c++20 datatype std::span offer the same functionality? You would have the advantage that you dont have to
reinterpret_cast every message to a char*.

**Comment by ghost** [comment]
Yep. Span is probably a better fit, but then you move from C++17 to C++20 and you also break APIs

**Reply by SpiRelli90** [reply]
Okay thats understandable. Maybe in the future...
